src/r5py/r5/transport_network.py

```diff
@@ -62,6 +64,47 @@ def __init__(self, osm_pbf, gtfs=[], build_config={}):
         )
         self._transport_network.transitLayer.buildDistanceTables(None)
 
+    @classmethod
+    def from_directory(cls, path):
+        """
+        Load a transport network, find input data in `path`.
+
+        This mimicks r5r’s behaviour to accept a directory path
+        as the only input to `setup_r5()`.
+
+        If more than one OpenStreetMap extract (`.osm.pbf`) is
+        found in `path`, the (alphabetically) first is used.
+        In case *no* OpenStreetMap extract is found, a `FileNotFound`
+        exception is raised. Any and all GTFS data files are used.
+
+        Arguments:
+        ----------
+        path : str
+            directory path in which to search for GTFS and .osm.pbf files
+
+        Returns:
+        --------
+        TransportNetwork
+            A fully initialised r5py.TransportNetwork
+        """
+        path = pathlib.Path(path)
+        try:
+            osm_pbf = sorted(path.glob("*.osm.pbf"))[0]
+        except KeyError:
+            raise FileNotFoundError(
+                "Could not find any OpenStreetMap extract file (`.osm.pbf`) in {}".format(
+                    str(path.absolute())
+                )
+            )
+        gtfs = list(path.glob("*GTFS.zip")) + list(path.glob("GTFS*.zip"))
```

Comment:
I guess this is case sensitive, or not? If yes, then probably would be good to deal with files that have possibly been written in different ways (lowercase, uppercase, capital)

Reply:
That’s a good point and also a mental TODO list item of mine which I forgot about: I’m still going to check which files R5 and r5r consider GTFS data sets.

Reply:
Okay, so R5 runs a slightly more sophisticated check, and I’m going to implement something along the same lines: 

1. It checks for the file extension, 
2. if it’s `.zip`, check its content
3. if it contains `stop_times.txt`, consider it GTFS

https://github.com/conveyal/r5/blob/dev/src/main/java/com/conveyal/r5/transit/TransportNetwork.java#L246

This is still a bit of a heuristic approach, but I guess we would fail later, in case there’s a non-GTFS zip file containing `stop_times.txt`. The [reference standard](https://developers.google.com/transit/gtfs/reference#field_definitions) lists `agency.txt`, `stops.txt`, `routes.txt`, `trips.txt`, `stop_times.txt` as *required*. 

This opens up yet another question (but that might be an issue for the future): should we even go there, or rather leave this all to R5 by calling its `TransportNetwork.fromDirectory()` under the hood? The thinner `r5py` stays, the easier it will be to maintain, and it would help avoid redundant code.

Reply:
Hmm, good point. My initial opinion would be to leave this logic to R5 for obvious reasons that you said here. If  needed for some customization, we can always decide to implement this ourselves later.

Reply:
I added it for now, because I think it makes sense, but let’s keep it in mind that this is not set in stone.